app/queries/user_conversations_query.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class UserConversationsQuery
+  attr_reader :relation, :user_id
+
+  def initialize(user_id, relation = Conversation.all)
+    @relation = relation
+    @user_id = user_id
+  end
+
+  def call
+    relation.where('sender_id = ? OR receiver_id = ?', user_id, user_id)
+  end
```

Comment:
I think this should be called different, maybe `for_user(user)'. Also the user_id parameter has more sense if sending just to this method. That way you can remove the user_id from the class initialization.

In the future, if you need to query different Conversations that may not be related to an specific User, then it's not necessary to send it when initializing this query Object.

```suggestion
  def for_user(user)
    relation.where('sender_id = :user_id OR receiver_id = :user_id', user_id: user.id)
  end
```